File: Stack/Stack1.java
class StackNode
{
public int data;
public StackNode next;
}
class Stack1
{
public static StackNode top=null;
public static void push(int num)
{
StackNode t;
t=new StackNode();
t.data=num;
t.next=top;
top=t;
}
public static int pop()
{
int data;
data=top.data;
top=top.next;
return data;
}
public static boolean isEmpty()
{
return top==null;
}
public static void main(String args[])
{
int i;
for(i=0;i<=9;i++) push(i);
while(!isEmpty()) System.out.println(pop());
}
}

File: Stack/Stack1.py
class StackNode:
    def __init__(self):
        self.data=0
        self.next=None
top=0
def push(num):
    global top
    t=StackNode()
    t.data=num
    t.next=top
    top=t
def pop():
    global top
    data=top.data
    top=top.next
    return data
def isEmpty():
    global top
    return top==0
for i in range(10): push(i)
while not isEmpty(): print(pop())
